components/AppBar/tests/unit/AppBarWrappingTopLayoutGuideTests.swift

```diff
@@ -0,0 +1,193 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import XCTest
+import MaterialComponents.MaterialAppBar
+import MaterialComponents.MaterialUIMetrics
+
+class AppBarWrappingTopLayoutGuideTests: XCTestCase {
+
+  // MARK: No scroll view
+
+  func testNoScrollViewTopLayoutGuideEqualsBottomEdgeOfHeaderView() {
+    // Given
+    let contentViewController = UIViewController()
+    let container = MDCAppBarContainerViewController(contentViewController: contentViewController)
+    container.isTopLayoutGuideAdjustmentEnabled = true
+    container.appBar.headerViewController.headerView.minMaxHeightIncludesSafeArea = false
+    container.appBar.headerViewController.headerView.minimumHeight = 50
+    container.appBar.headerViewController.headerView.maximumHeight = 100
+    let _ = container.view // Force the view to load.
+
+    // Then
+    XCTAssertEqual(contentViewController.view.autoresizingMask, [.flexibleWidth, .flexibleHeight])
+    XCTAssertEqual(contentViewController.view.frame, container.view.bounds)
+    XCTAssertEqual(contentViewController.topLayoutGuide.length,
+                   container.appBar.headerViewController.headerView.frame.maxY)
+    #if swift(>=3.2)
+    if #available(iOS 11.0, *) {
+      XCTAssertEqual(contentViewController.additionalSafeAreaInsets.top,
+                     container.appBar.headerViewController.headerView.frame.maxY
+                      - MDCDeviceTopSafeAreaInset())
```

Comment:
Given that this inspects the device's keywindow, will the AppBar layout break if it's not positioned at the topmost part of the screen or if it's in a non-key window? For example, on a secondary screen?

Reply:
Currently the App Bar always assumes that it is part of the key window unless statusBarHintCanOverlapHeader is changed.

Reply:
Ack.